### PR TITLE
ARROW-9507: [Rust] [DataFusion] Implement Display for PhysicalExpr

### PR DESCRIPTION
**Pull request by andygrove**
This allows physical expressions to be printed in a human-readable form and is a step towards a printable physical plan.

### CODE REVIEW

**Comment by andygrove** [comment]
fyi @jorgecarleitao

File: rust/datafusion/src/execution/physical_plan/udf.rs [hunk 1]
```diff
@@ -84,6 +86,19 @@ impl ScalarFunctionExpr {
     }
 }
 
+impl fmt::Display for ScalarFunctionExpr {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}(", self.name)?;
+        for i in 0..self.args.len() {
```
Comment (jorgecarleitao) [inline rust/datafusion/src/execution/physical_plan/udf.rs:92]: `self.args.join(", ")`?

Comment (jorgecarleitao) [reply rust/datafusion/src/execution/physical_plan/udf.rs:92]: Or, the complete statement, `write!(f, "{}({})", self.name, self.args.join(", "))?;`

**Comment by jorgecarleitao** [comment]
Left a minor comment, other than that, LGTM. 

I am uncertain whether this should be Debug or Display, but since we haven't formulated a concrete spec on this, Display is fine for now. :)

File: rust/datafusion/src/execution/physical_plan/udf.rs (hunk 1 above)
Comment (andygrove) [reply rust/datafusion/src/execution/physical_plan/udf.rs]: That wouldn't work but I pushed this change instead:

```rust
        write!(
            f,
            "{}({})",
            self.name,
            self.args
                .iter()
                .map(|e| format!("{}", e))
                .collect::<Vec<String>>()
                .join(", ")
        )
```

Comment (jorgecarleitao) [reply rust/datafusion/src/execution/physical_plan/udf.rs]: yeap, forgot that .join required strings. Still better IMO, as it is more expressive.

**Comment by kszucs** [comment]
@andygrove please rebase

**Pull request closed without merging**